### ISSUE TITLE
Check if saved Bookmark URL is still avaiable

### ISSUE THREAD

**Issue by schneekluth** [shaarli/Shaarli#1256]
I've been using Shaarli quite some time and have about 1000 bookmarks in the data store. From time to time I find some of them that are not available anymore. It would be nice to have an option in the settings to check all stored links for availability and have a cleanup option to delete the ones unavailable. Or maybe just a section in the docs with a little tutorial how to do this from CLI.

**Comment by nodiscc** [comment]
The problem is how you determine with certainty that a link is "not available anymore"?

 - Some hosts will return a proper 404 error so it may be safe to assume that the page is gone.
 - Some hosts will return a generic error page while still serving a HTTP code 200. The actual error may or may not be explained on the page. What to do then?
 - Some hosts will not reply at all, possibly because the whole site is gone, or because of a temporary/network problem. How to guess?

**Comment by schneekluth** [comment]
@nodiscc At least case 1 and 3 would be an improvement. Case 2 with code 200 is of course not possible without human verification.

**Comment by kalvn** [comment]
It could work in two steps.

After launching the analysis, the list of all 404 that were deleted is displayed with an option to explicitly keep some of the sites listed. Then the list of all sites matching your cases 2 and 3 are listed separately in order to allow manual verification with a button to either keep (ignore) or delete each site.  And maybe a red button to delete everything, for people in a hurry.

To sum up, if the user does nothing else than starting the analysis, all sites matching case 1 are deleted, all others are kept.

**Comment by lmjst** [comment]
Hi,
I'm also looking for this links verification tool. 
With https://validator.w3.org/checklink I succeeded in testing the public links (needed to amend the robots.txt with allowed user agent  W3C-checklink)
